modules/tinymce/src/core/main/ts/lookup/UserLookup.ts

```diff
@@ -134,23 +194,28 @@ const UserLookup = (editor: Editor): UserLookup => {
         pendingResolvers.delete(userId);
       });
 
-  const finallyResolve = (userId: UserId, user: User) =>
+  const finallyResolve = (userId: UserId, user: ExpectedUser) =>
     Optional
       .from(pendingResolvers.get(userId))
       .each(({ resolve }) => {
-        resolve(user);
+        resolve({
+          ...user,
+          id: user.id,
+          name: Optional.from(user.name).getOr(user.id),
+          avatar: Optional.from(user.avatar).getOr(deriveAvatar(user.id)),
```

Comment:
Doing null checks with options is no the way. We use options to avoid null yes, but if your expression checks for null and returns non null then null is not spread though the whole code base it's local.

```suggestion
          name: user.name ?? user.id,
          avatar: user.avatar ?? deriveAvatar(user.id),
```

Reply:
We don't even need to treat these as optionals or check for null values as we already handle assigning them fallbacks in `validateResponse` if they are not present so I think we can resolve this one.